File: lib-base/src/main/java/com/llj/lib/base/MvcBaseToolbarActivity.java
package com.llj.lib.base;

import androidx.constraintlayout.widget.ConstraintLayout;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.llj.lib.base.config.ToolbarConfig;

/**
 * ArchitectureDemo.
 * describe:
 *
 * @author llj
 * @date 2019-07-25
 */
public abstract class MvcBaseToolbarActivity extends MvcBaseActivity {
    protected ConstraintLayout mClToolbar;
    protected ImageView        mIvTbClose;
    protected TextView         mTvTbTitle;
    protected TextView         mTvTbRight;
    protected ImageView        mIvTbRight;
    protected View             mVDivide;

    @Override
    public View layoutView() {
        ViewGroup mRootView = (ViewGroup) getLayoutInflater().inflate(R.layout.mvc_base_title_activity, null);
        if (layoutId() != 0) {
            mClToolbar = mRootView.findViewById(R.id.cl_toolbar);
            mIvTbClose = mRootView.findViewById(R.id.iv_close);
            mTvTbTitle = mRootView.findViewById(R.id.tv_title);
            mTvTbRight = mRootView.findViewById(R.id.tv_right);
            mIvTbRight = mRootView.findViewById(R.id.iv_right);
            mVDivide = mRootView.findViewById(R.id.v_divide);
            initToolbar();
            getLayoutInflater().inflate(layoutId(), mRootView, true);
        }
        return mRootView;
    }

    private void initToolbar() {
        ToolbarConfig toolbarConfig = AppManager.getInstance().getToolbarConfig();
        if (toolbarConfig != null) {
            if (toolbarConfig.getToolbarBackgroundColorRes() >= 0) {
                mClToolbar.setBackgroundColor(getCompatColor(this, toolbarConfig.getToolbarBackgroundColorRes()));
            }
            if (toolbarConfig.getToolbarBackgroundDrawable() != null) {
                mClToolbar.setBackground(toolbarConfig.getToolbarBackgroundDrawable());
            }

            mIvTbClose.setImageResource(toolbarConfig.getLeftImageRes());

            mTvTbTitle.setTextSize(toolbarConfig.getTitleUnit(), toolbarConfig.getTitleSize());
            if (toolbarConfig.getTitleColorRes() >= 0) {
                mTvTbTitle.setTextColor(getCompatColor(this, toolbarConfig.getTitleColorRes()));
            }

            if (toolbarConfig.getRightImageRes() >= 0) {
                mIvTbRight.setImageResource(toolbarConfig.getRightImageRes());
            }

            mTvTbRight.setTextSize(toolbarConfig.getRightUnit(), toolbarConfig.getRightTextSize());
            if (toolbarConfig.getRightTextColorRes() > 0) {
                mTvTbRight.setTextColor(getCompatColor(this, toolbarConfig.getRightTextColorRes()));
            }

            ConstraintLayout.LayoutParams layoutParams2 = (ConstraintLayout.LayoutParams) mVDivide.getLayoutParams();
            layoutParams2.height = toolbarConfig.getDivideHeight();
            if (toolbarConfig.getDivideBackgroundColorRes() >= 0) {
                mVDivide.setBackgroundColor(getCompatColor(this, toolbarConfig.getDivideBackgroundColorRes()));
            }
            mVDivide.setVisibility(toolbarConfig.getVisibility());
        }

    }


}

File: lib-base/src/main/java/com/llj/lib/base/MvcBaseToolbarActivity.kt
package com.llj.lib.base

import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.viewbinding.ViewBinding

/**
 * ArchitectureDemo.
 * describe:
 *
 * @author llj
 * @date 2019-07-25
 */
abstract class MvcBaseToolbarActivity<V : ViewBinding> : MvcBaseActivity<V>() {

  lateinit var mClToolbar: ConstraintLayout
  lateinit var mIvTbClose: ImageView
  lateinit var mTvTbTitle: TextView
  lateinit var mTvTbRight: TextView
  lateinit var mIvTbRight: ImageView
  lateinit var mVDivide: View


  override fun layoutId(): Int {
    return 0
  }

  override fun layoutView(): View? {
    val mRootView = layoutInflater.inflate(R.layout.mvc_base_title_activity, null) as ViewGroup

    applyNavigationInsets(mRootView)

    if (layoutId() != 0) {
      initToolbar(mRootView)
      layoutInflater.inflate(layoutId(), mRootView, true)
    } else {
      var layoutViewBinding = layoutViewBinding()
      if (layoutViewBinding == null) {
        //如果忘记设置则使用反射设置
        layoutViewBinding = reflectViewBinder(layoutInflater, mTagLog)
      }
      if (layoutViewBinding != null) {
        mViewBinder = layoutViewBinding
        mRootView.addView(mViewBinder.root, LinearLayout.LayoutParams(-1, -1))
      }
    }
    return mRootView
  }

  private fun initToolbar(view: View) {

    mClToolbar = view.findViewById(R.id.cl_toolbar)
    mIvTbClose = view.findViewById(R.id.iv_close)
    mTvTbTitle = view.findViewById(R.id.tv_title)
    mTvTbRight = view.findViewById(R.id.tv_right)
    mIvTbRight = view.findViewById(R.id.iv_right)
    mVDivide = view.findViewById(R.id.v_divide)


    //根据配置文件设置toolbar
    val toolbarConfig = AppManager.getInstance().toolbarConfig
    if (toolbarConfig != null) {
      if (toolbarConfig.toolbarBackgroundColorRes >= 0) {
        mClToolbar.setBackgroundColor(getCompatColor(this, toolbarConfig.toolbarBackgroundColorRes))
      }
      if (toolbarConfig.toolbarBackgroundDrawable != null) {
        mClToolbar.background = toolbarConfig.toolbarBackgroundDrawable
      }
      mIvTbClose.setImageResource(toolbarConfig.leftImageRes)
      mTvTbTitle.setTextSize(toolbarConfig.titleUnit, toolbarConfig.titleSize.toFloat())
      if (toolbarConfig.titleColorRes >= 0) {
        mTvTbTitle.setTextColor(getCompatColor(this, toolbarConfig.titleColorRes))
      }
      if (toolbarConfig.rightImageRes >= 0) {
        mIvTbRight.setImageResource(toolbarConfig.rightImageRes)
      }
      mTvTbRight.setTextSize(toolbarConfig.rightUnit, toolbarConfig.rightTextSize.toFloat())
      if (toolbarConfig.rightTextColorRes > 0) {
        mTvTbRight.setTextColor(getCompatColor(this, toolbarConfig.rightTextColorRes))
      }
      val layoutParams2 = mVDivide!!.layoutParams as ConstraintLayout.LayoutParams
      layoutParams2.height = toolbarConfig.divideHeight
      if (toolbarConfig.divideBackgroundColorRes >= 0) {
        mVDivide.setBackgroundColor(getCompatColor(this, toolbarConfig.divideBackgroundColorRes))
      }
      mVDivide.visibility = toolbarConfig.visibility
    }
  }
}